### PR TITLE
perf(db): Fine-tune state keeper cache performance / RAM usage

### PR DESCRIPTION
**Pull request by slowli**
## What ❔

Allows to configure state keeper cache performance / RAM usage on EN.

## Why ❔

Right now, state keeper cache uses very small block cache (order of 5 MB), and eventually loads ~8 GB of SST files into the page cache. Both can be configured more optimally (by increasing block cache size and setting max open files for RocksDB).

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.
- [x] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/protobuf_config/src/experimental.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        let state_keeper_db_block_cache_capacity_mb =
+            *required(&self.state_keeper_db_block_cache_capacity_mb)
+                .context("state_keeper_db_block_cache_capacity_mb")?;
+        Ok(configs::ExperimentalDBConfig {
+            state_keeper_db_block_cache_capacity_mb: state_keeper_db_block_cache_capacity_mb
+                .try_into()
+                .context("state_keeper_db_block_cache_capacity_mb")?,
+            state_keeper_db_max_open_files: self
+                .state_keeper_db_max_open_files
+                .and_then(NonZeroU32::new),
```
Comment (pompon0) [inline core/lib/protobuf_config/src/experimental.rs:22]: context(...)

File: core/lib/protobuf_config/src/experimental.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+use crate::proto::experimental as proto;
+
+impl ProtoRepr for proto::Db {
+    type Type = configs::ExperimentalDBConfig;
+
+    fn read(&self) -> anyhow::Result<Self::Type> {
+        let state_keeper_db_block_cache_capacity_mb =
+            *required(&self.state_keeper_db_block_cache_capacity_mb)
+                .context("state_keeper_db_block_cache_capacity_mb")?;
```
Comment (pompon0) [inline core/lib/protobuf_config/src/experimental.rs:15]: *required(...).and_then(|x|Ok(x.try_into()?)).context(...)

File: core/lib/protobuf_config/src/proto/config/experimental.proto [hunk 3]
```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
```
Comment (pompon0) [inline core/lib/protobuf_config/src/proto/config/experimental.proto:1]: add a comment that this file is excluded from the compatibility check

File: core/lib/protobuf_config/src/database.rs [hunk 4]
```diff
@@ -66,19 +66,22 @@ impl ProtoRepr for proto::MerkleTree {
...
 impl ProtoRepr for proto::Db {
     type Type = configs::database::DBConfig;
+
     fn read(&self) -> anyhow::Result<Self::Type> {
         Ok(Self::Type {
             state_keeper_db_path: required(&self.state_keeper_db_path)
                 .context("state_keeper_db_path")?
                 .clone(),
             merkle_tree: read_required_repr(&self.merkle_tree).context("merkle_tree")?,
+            experimental: read_required_repr(&self.experimental).context("experimental")?,
```
Comment (pompon0) [inline core/lib/protobuf_config/src/database.rs:76]: nit: are you sure you want this field to be required?

**Review by pompon0 — DISMISSED** [review]

**Review by RomanBrodetski — APPROVED** [review]

**Review by pompon0 — APPROVED** [review]

**Pull request merged**